fuel-block-producer/src/block_producer/tests.rs

```diff
@@ -0,0 +1,231 @@
+use crate::{
+    mocks::{
+        FailingMockExecutor,
+        MockDb,
+        MockExecutor,
+        MockRelayer,
+        MockTxPool,
+    },
+    Config,
+    Producer,
+};
+use fuel_core_interfaces::{
+    block_producer::{
+        BlockProducer,
+        Error,
+    },
+    executor::Executor,
+    model::{
+        FuelBlockDb,
+        FuelBlockHeader,
+    },
+};
+use rand::{
+    rngs::StdRng,
+    Rng,
+    SeedableRng,
+};
+use std::sync::{
+    Arc,
+    Mutex,
+};
+
+#[tokio::test]
+async fn cant_produce_at_genesis_height() {
+    let ctx = TestContext::default();
+    let producer = ctx.producer();
+
+    let err = producer
+        .produce_block(0u32.into())
+        .await
+        .expect_err("expected failure");
+
+    assert!(
+        matches!(err.downcast_ref::<Error>(), Some(Error::GenesisBlock)),
+        "unexpected err {:?}",
+        err
+    );
+}
+
+#[tokio::test]
+async fn can_produce_initial_block() {
+    // simple happy path for producing the first block
+    let ctx = TestContext::default();
+    let producer = ctx.producer();
+
+    let err = producer
+        .produce_block(0u32.into())
```

Comment:
I guess it should be `1`? And test should be positive?

Reply:
oops copy paste bug

Reply:
fixed